src/main/java/com/hivemq/cli/commands/options/TlsOptions.java

```
@@ -138,104 +127,255 @@ private boolean useBuiltSslConfig() {
-            i++;
-        }
+    /**
+     * The truststore loading uses a natural hierarchy to determine precedence over the security provider possibilities.
+     * <ol>
+     *     <li>A truststore provided as input parameter</li>
+     *     <li>A server/ca certificate provided as input parameter</li>
+     *     <li>A truststore provided as default option inside the cli configuration file</li>
+     *     <li>A server/ca certificate provided as default option inside the cli configuration file</li>
+     * </ol>
+     *
+     * @return a trust manager factory or null if no trust provider was configured.
+     * @throws Exception if a truststore or certificate could not be accessed.
+     */
+    private @Nullable TrustManagerFactory buildTrustManagerFactory() throws Exception {
+        final DefaultCLIProperties defaultCLIProperties =
+                Objects.requireNonNull(MqttCLIMain.MQTTCLI).defaultCLIProperties();
+        final TrustManagerFactory trustManagerFactory;
 
-        final TrustManagerFactory trustManagerFactory =
-                TrustManagerFactory.getInstance(TrustManagerFactory.getDefaultAlgorithm());
+        final Path defaultTruststore = defaultCLIProperties.getTruststore();
+        final Path defaultServerCertificate = defaultCLIProperties.getServerCertificateChain();
 
-        trustManagerFactory.init(ks);
+        final boolean truststoreExists = clientTruststorePath != null;
+        final boolean certificatesExists = serverCertificatePath != null || serverCertificateDirPath != null;
+        final boolean defaultTruststoreExists = defaultTruststore != null;
+        final boolean defaultCertificateExists = defaultServerCertificate != null;
+
+        if (truststoreExists) {
+            if (certificatesExists) {
+                Log.warn("The truststore parameter shadows the server/ca certificate parameter. " +
+                        "If you want to use them instead please remove the keystore parameter.");
```

I would throw an exception here.